src/service/viewer-impl.js

```diff
@@ -143,6 +143,20 @@ export class Viewer {
     /** @const @private {!Object<string, string>} */
     this.params_ = {};
 
+    /** @private {?function()} */
+    this.whenVisibleResolve_ = null;
+
+    /** @private @const {!Promise} */
+    this.whenVisiblePromise_ = new Promise((resolve) => {
+      this.whenVisibleResolve_ = resolve;
+    });
+
+    // Immediately resolve the promise if by the time the `Viewer`
+    // instance is constructed, the document is already `visible`.
+    if (this.isVisible()) {
```

Comment:
done. moved at the end of the constructor and i just directly call `this.onVisibilityChange_()`.